arduino/smartcar/smartcar.ino

```diff
@@ -1,9 +1,80 @@
 #include <Smartcar.h>
 
-void setup() {
-  // put your setup code here, to run once:
+const int fSpeed   = 70;  
+const int bSpeed   = -70; 
+const int lDegrees = -75; 
+const int rDegrees = 75;  
+const int TRIGGER_PIN = 6;
+const int ECHO_PIN = 7;
+const unsigned int MAX_DISTANCE =100;
+
+ArduinoRuntime arduinoRuntime;
+SR04 front(arduinoRuntime, TRIGGER_PIN, ECHO_PIN, MAX_DISTANCE);
+BrushedMotor leftMotor(arduinoRuntime, smartcarlib::pins::v2::leftMotorPins);
+BrushedMotor rightMotor(arduinoRuntime, smartcarlib::pins::v2::rightMotorPins);
+DifferentialControl control(leftMotor, rightMotor);
+
+SimpleCar car(control);
+
+void setup()
+{
+    Serial.begin(9600);
+}
+
+void handleInput(){
+    if(Serial.available())
+    {
+        char input = Serial.read(); 
+                                    
+        switch (input)
+        {
+        case 'l': 
+            car.setSpeed(fSpeed);
+            car.setAngle(lDegrees);
+            break;
+        case 'r': 
+            car.setSpeed(fSpeed);
+            car.setAngle(90);
+            break;
+        case 'f': 
+            car.setSpeed(fSpeed);
+            car.setAngle(0);
+            break;
+        case 'b': 
+            car.setSpeed(bSpeed);
+            car.setAngle(0);
+            break;
+        default: 
+            car.setSpeed(0);
+            car.setAngle(0);
+        }
+    }
 }
 
-void loop() {
-  // put your main code here, to run repeatedly:
+void loop()
+{
+   
+    Serial.println(front.getDistance());
+    delay(100);
+    
+    if(front.getDistance() <= 100 && front.getDistance()>0){
```

Comment:
It's not a good practice to start a new pull request because we lose the previous conversations. Don't be afraid of `git`. :)

Anyway, the problem remains. Read the question carefully and look at L60:

> What happens if the first call to getDistance returns 0 and the next one 150?

Reply:
I talked to another group regarding the CI comment that you sended. They told me that instead of sending a new file I should change the smartcar file in the repository because that one uses CI so I did that. I also kept the other pull request so that the conversation could still be there. 

anyway, regarding the question so you mean that my latest response is not enough?

"No, the car should not stop then, if we receive 0 then there no obstacle and the car does not stops. if we get 150 then the car will still not stops because is not inside the range. The car only stops if getDistance is between 1-100. ( less or equal to 100 or bigger than 0)

But I dont understand your comment " What happens if the first call to getDistance returns 0 and the next one 150? **The car stops while it should not**."

Reply:
> They told me that instead of sending a new file I should change the smartcar file in the repository because that one uses CI so I did that

Or you could change the CI so the new file is built instead (or both!). :)

> "No, the car should not stop then, if we receive 0 then there no obstacle and the car does not stops. if we get 150 then the car will still not stops because is not inside the range. The car only stops if getDistance is between 1-100. ( less or equal to 100 or bigger than 0)"

You are mistaken. That is **not** what the car will do, that is what you _wish_ it would.
Read the line of code you have carefully and then my question:

> What happens if the **first** call to getDistance returns 0 and the **next** one 150?

Ask the rest of the @DIT113-V22/group-08 for help and further input. If I say anything more I will be basically giving away the solution which is rather simple.

Reply:
Or do you mean that both measuerement are being evaluated at the same time ?
In that case,  I have two measurements in the if statement from the sensor, that both could have different values at some point . In that case  it may be a better choice to create a variable to store the measurement and ensure that the if statement evaluates both conditions with the same measurement?

Reply:
:+1:

Reply:
Two consecutive measurements, no matter how close they happen, are not guaranteed to give you the same value. The measurements (especially in real life) are noisy. The scenario I am describing is very likely.